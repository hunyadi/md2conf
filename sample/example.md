<!-- confluence-page-id: 85668266616 -->

If you are a user who wants to publish pages to Confluence, you should install the package [markdown-to-confluence](https://pypi.org/project/markdown-to-confluence/) from PyPI. If you are a developer who wants to contribute, you should clone the repository [md2conf](https://github.com/hunyadi/md2conf) from GitHub.

# Publish Markdown to Confluence

This page demonstrates the various features of converting a Markdown document into an XHTML document compliant with the *Confluence Storage Format*.

## Basic features

Text with **bold**, *italic*, `monospace`, <ins>underline</ins> and ~~strikethrough~~.

Absolute link to an [external location](http://example.com/) or relative link to a [heading in the same document](#Basic-features). Relative links to other pages are also supported if you are synchronizing a directory tree, not a single file. "Naked" links such as URLs and e-mails are automatically identified: <https://example.com> and <mailto:me@example.com>. Relative URLs to [locations not exported](missing.md) may be skipped.

An ordered list:

1. *List item 1*
2. **List item 2**
spanning multiple text file lines but rendered as a single line
3. _List item 3_
4. `List item 4`

A bullet point list:

* *List item 1*
* **List item 2**
spanning multiple text file lines but rendered as a single line
* _List item 3_
* `List item 4`

An unordered list:

- *List item 1*
- **List item 2**
spanning multiple text file lines but rendered as a single line
- _List item 3_
- `List item 4`

A nested list:

* List item 1
    * Nested item 1
        1. Item 1
        2. Item 2
        3. Item 3
    * Nested item 2
        - Item 4
        - Item 5
        - Item 6
* List item 2
    1. Nested item 3
    2. Nested item 4

![PNG image with caption](figure/interoperability.png)

![SVG image with caption](figure/interoperability.svg)

## Code blocks

A language-neutral code block:
```
func:
    preformatted text
```

A code block in Python:
```python
def func(n: int) -> str:
    return str(n)
```

A code block in Java:
```java
class Simple {
    public static void main(String args[]) {
        System.out.println("Hello Java!");
    }
}
```

A code block in JSON:
```json
{
    "boolean": true,
    "integer": 42,
    "string": "value",
    "list": [1,2,3]
}
```

## Sections

Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam dignissim sem vel tortor scelerisque sodales. Quisque id faucibus massa. Integer enim nulla, cursus vel facilisis id, convallis in nunc. Suspendisse potenti. Aenean vehicula semper fringilla. Nulla pretium vitae enim sed feugiat. Integer molestie a elit ac fermentum.

### Subsection 1

Integer vel cursus neque. Ut a ex orci. Nulla elementum lacus sed accumsan rutrum. Donec dui augue, porta at diam quis, ultricies rhoncus ipsum. Aliquam posuere mollis varius. Ut tempor velit facilisis ligula mattis consectetur. Praesent sit amet enim vitae est facilisis auctor vitae at nisi. Aliquam erat volutpat. Donec commodo vestibulum lorem sit amet semper. Vivamus vestibulum sem bibendum, pretium risus aliquam, pharetra augue. Donec id quam sit amet est cursus elementum ut eget massa. Duis vel sapien nunc. Phasellus dui risus, auctor eu erat vitae, vehicula dignissim justo. Morbi nec auctor leo. Praesent tincidunt faucibus tortor in iaculis.

### Subsection 2

Nullam a dictum urna, volutpat convallis purus. Fusce at vulputate mauris. Donec dignissim id nisl quis ultricies.
Quisque pulvinar nulla risus, vel tincidunt felis pretium hendrerit. Pellentesque placerat, mi vel luctus tristique,
sem nisi volutpat dolor, vitae pulvinar ipsum orci vitae ligula. Morbi tempus at tortor sed volutpat. Vivamus
facilisis orci libero, ac fermentum metus iaculis eget. Morbi aliquam enim et semper dapibus. Suspendisse ut velit sit
amet metus convallis elementum. Cras sagittis justo ac venenatis pulvinar. Integer est massa, blandit vel nulla at,
dignissim finibus dui.

## Admonitions

Markdown has no native support for admonitions. Admonitions that follow the [Python-Markdown syntax](https://python-markdown.github.io/extensions/admonition/) are converted into Confluence structured macros, which are shown as info panels.

!!! info
    This is an information panel.

!!! info "Optional explicit title"
    This is an information panel with an explicit title.

    This is the second paragraph with a [link](https://example.com/).

!!! tip "Tip of the day"
    This is a structured macro panel showing a tip.

!!! note "A note"
    This is a structured macro panel showing a note.

!!! warning "A warning message"
    This is a structured macro panel showing a warning.

## UTF-8 characters

UTF-8 characters outside of the ASCII range are also supported: e.g. árvíztűrő tükörfúrógép.
