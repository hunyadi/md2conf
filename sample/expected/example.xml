<ac:structured-macro ac:name="info" ac:schema-version="1"><ac:rich-text-body><p>This page has been generated with a tool.</p></ac:rich-text-body></ac:structured-macro><p>If you are a user who wants to publish pages to Confluence, you should install the package <a href="https://pypi.org/project/markdown-to-confluence/">markdown-to-confluence</a> from PyPI. If you are a developer who wants to contribute, you should clone the repository <a href="https://github.com/hunyadi/md2conf">md2conf</a> from GitHub.</p><h1>Publish Markdown to Confluence</h1><p>This page demonstrates the various features of converting a Markdown document into an XHTML document compliant with the <em>Confluence Storage Format</em>.</p><h2>Basic features</h2><p>Text with <strong>bold</strong>, <em>italic</em>, <code>monospace</code>, <ins>underline</ins> and <del>strikethrough</del>.</p><p>Absolute link to an <a href="http://example.com/">external location</a> or relative link to a <a href="#Basic-features">heading in the same document</a>. Relative links to other pages are also supported if you are synchronizing a directory tree, not a single file. "Naked" links such as URLs and e-mails are automatically identified: <a href="https://example.com">https://example.com</a> and <a href="mailto:me@example.com">me@example.com</a>. Relative URLs to <a>locations not exported</a> may be skipped.</p><p>An ordered list:</p><ol><li><em>List item 1</em></li><li><strong>List item 2</strong> spanning multiple text file lines but rendered as a single line</li><li><em>List item 3</em></li><li><code>List item 4</code></li></ol><p>A bullet point list:</p><ul><li><em>List item 1</em></li><li><strong>List item 2</strong> spanning multiple text file lines but rendered as a single line</li><li><em>List item 3</em></li><li><code>List item 4</code></li></ul><p>An unordered list:</p><ul><li><em>List item 1</em></li><li><strong>List item 2</strong> spanning multiple text file lines but rendered as a single line</li><li><em>List item 3</em></li><li><code>List item 4</code></li></ul><p>A nested list:</p><ul><li>List item 1<ul><li>Nested item 1<ol><li>Item 1</li><li>Item 2</li><li>Item 3</li></ol> </li><li>Nested item 2<ul><li>Item 4</li><li>Item 5</li><li>Item 6</li></ul> </li></ul> </li><li>List item 2<ol><li>Nested item 3</li><li>Nested item 4</li></ol> </li></ul><ac:image ac:align="center" ac:layout="center"><ri:attachment ri:filename="figure_interoperability.png"/><ac:caption><p>PNG image with caption</p></ac:caption></ac:image><ac:image ac:align="center" ac:layout="center"><ri:attachment ri:filename="figure_interoperability.png"/><ac:caption><p>SVG image with caption</p></ac:caption></ac:image><h2>Code blocks</h2><p>A language-neutral code block:</p><ac:structured-macro ac:name="code" ac:schema-version="1"><ac:parameter ac:name="theme">Midnight</ac:parameter><ac:parameter ac:name="language">none</ac:parameter><ac:parameter ac:name="linenumbers">true</ac:parameter><ac:plain-text-body><![CDATA[func:
    preformatted text]]></ac:plain-text-body></ac:structured-macro><p>A code block in Python:</p><ac:structured-macro ac:name="code" ac:schema-version="1"><ac:parameter ac:name="theme">Midnight</ac:parameter><ac:parameter ac:name="language">python</ac:parameter><ac:parameter ac:name="linenumbers">true</ac:parameter><ac:plain-text-body><![CDATA[def func(n: int) -> str:
    return str(n)]]></ac:plain-text-body></ac:structured-macro><p>A code block in Java:</p><ac:structured-macro ac:name="code" ac:schema-version="1"><ac:parameter ac:name="theme">Midnight</ac:parameter><ac:parameter ac:name="language">java</ac:parameter><ac:parameter ac:name="linenumbers">true</ac:parameter><ac:plain-text-body><![CDATA[class Simple {
    public static void main(String args[]) {
        System.out.println("Hello Java!");
    }
}]]></ac:plain-text-body></ac:structured-macro><p>A code block in JSON:</p><ac:structured-macro ac:name="code" ac:schema-version="1"><ac:parameter ac:name="theme">Midnight</ac:parameter><ac:parameter ac:name="language">json</ac:parameter><ac:parameter ac:name="linenumbers">true</ac:parameter><ac:plain-text-body><![CDATA[{
    "boolean": true,
    "integer": 42,
    "string": "value",
    "list": [1,2,3]
}]]></ac:plain-text-body></ac:structured-macro><h2>Sections</h2><p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam dignissim sem vel tortor scelerisque sodales. Quisque id faucibus massa. Integer enim nulla, cursus vel facilisis id, convallis in nunc. Suspendisse potenti. Aenean vehicula semper fringilla. Nulla pretium vitae enim sed feugiat. Integer molestie a elit ac fermentum.</p><h3>Subsection 1</h3><p>Integer vel cursus neque. Ut a ex orci. Nulla elementum lacus sed accumsan rutrum. Donec dui augue, porta at diam quis, ultricies rhoncus ipsum. Aliquam posuere mollis varius. Ut tempor velit facilisis ligula mattis consectetur. Praesent sit amet enim vitae est facilisis auctor vitae at nisi. Aliquam erat volutpat. Donec commodo vestibulum lorem sit amet semper. Vivamus vestibulum sem bibendum, pretium risus aliquam, pharetra augue. Donec id quam sit amet est cursus elementum ut eget massa. Duis vel sapien nunc. Phasellus dui risus, auctor eu erat vitae, vehicula dignissim justo. Morbi nec auctor leo. Praesent tincidunt faucibus tortor in iaculis.</p><h3>Subsection 2</h3><p>Nullam a dictum urna, volutpat convallis purus. Fusce at vulputate mauris. Donec dignissim id nisl quis ultricies. Quisque pulvinar nulla risus, vel tincidunt felis pretium hendrerit. Pellentesque placerat, mi vel luctus tristique, sem nisi volutpat dolor, vitae pulvinar ipsum orci vitae ligula. Morbi tempus at tortor sed volutpat. Vivamus facilisis orci libero, ac fermentum metus iaculis eget. Morbi aliquam enim et semper dapibus. Suspendisse ut velit sit amet metus convallis elementum. Cras sagittis justo ac venenatis pulvinar. Integer est massa, blandit vel nulla at, dignissim finibus dui.</p><h2>Admonitions</h2><p>Markdown has no native support for admonitions. Admonitions that follow the <a href="https://python-markdown.github.io/extensions/admonition/">Python-Markdown syntax</a> are converted into Confluence structured macros, which are shown as info panels.</p><ac:structured-macro ac:name="info" ac:schema-version="1"><ac:parameter ac:name="title">Info</ac:parameter><ac:rich-text-body><p>This is an information panel.</p></ac:rich-text-body></ac:structured-macro><ac:structured-macro ac:name="info" ac:schema-version="1"><ac:parameter ac:name="title">Optional explicit title</ac:parameter><ac:rich-text-body><p>This is an information panel with an explicit title.</p><p>This is the second paragraph with a <a href="https://example.com/">link</a>.</p></ac:rich-text-body></ac:structured-macro><ac:structured-macro ac:name="tip" ac:schema-version="1"><ac:parameter ac:name="title">Tip of the day</ac:parameter><ac:rich-text-body><p>This is a structured macro panel showing a tip.</p></ac:rich-text-body></ac:structured-macro><ac:structured-macro ac:name="note" ac:schema-version="1"><ac:parameter ac:name="title">A note</ac:parameter><ac:rich-text-body><p>This is a structured macro panel showing a note.</p></ac:rich-text-body></ac:structured-macro><ac:structured-macro ac:name="warning" ac:schema-version="1"><ac:parameter ac:name="title">A warning message</ac:parameter><ac:rich-text-body><p>This is a structured macro panel showing a warning.</p></ac:rich-text-body></ac:structured-macro><h2>UTF-8 characters</h2><p>UTF-8 characters outside of the ASCII range are also supported: e.g. árvíztűrő tükörfúrógép.</p>